src/wallet/spend.cpp

```diff
@@ -470,27 +468,28 @@ std::optional<SelectionResult> SelectCoins(const CWallet& wallet, const std::vec
             input_bytes = CalculateMaximumSignedInputSize(txout, &coin_control.m_external_provider, /* use_max_sig */ true);
         }
 
-        CInputCoin coin(outpoint, txout, input_bytes);
-        if (coin.m_input_bytes == -1) {
+        /* Set some defaults for depth, spendable, solvable, safe, time, and from_me as these don't matter for preset inputs since no selection is being done. */
+        COutput output(outpoint, txout, /*depth=*/ 0, input_bytes, /*spendable=*/ true, /*solvable=*/ true, /*safe=*/ true, /*time=*/ 0, /*from_me=*/ false);
+        if (output.m_input_bytes == -1) {
```

Comment:
Done

Reply:
Actually we still need this check here because `input_bytes` could be set by either `CalculateMaximumSignedInputSize` or `GetVirtualTransactionSize`, depending on whether the input weight is provided (this may have been a hidden merge conflict though so not apparent in the diff).